### PR TITLE
feat(textbox): prefix & suffix

### LINKED ISSUE #2191: Ebay-textbox: add prefix property in addition to leading icon
## Summary
The textbox component should have a prefix and postfix property available that accept characters in addition to the pre- and postfix icons. This allows for entering values like currencies and measurements, like AU$ or cm.

## Specs 
image

image

### PR DESCRIPTION
**Pull request by LuLaValva**
- Fixes #2191

- [x] This PR contains CSS changes
- [ ] This PR does not contain CSS changes

## Description

- Add prefix and suffix text to textbox
- Refactored textbox to use flex
  - Markup not modified
  - `textbox--icon-end` no longer required
  - *Until full `:has` support*, `textbox` needs `--disabled`, `--readonly`, and `--invalid` modifiers

## Notes

- **Potential regression:** clicking the icon no longer focuses on the textbox

## Screenshots

image

## Checklist

- [x] I verify the build is in a non-broken state
- [ ] I verify all changes are within scope of the linked issue

- [x] I regenerated all CSS files under dist folder
- [ ] I tested the UI in all supported browsers
- [ ] I did a visual regression check of the components impacted by doing a Percy build and approved the build
- [x] I tested the UI in dark mode and RTL mode
- [x] I added/updated/removed Storybook coverage as appropriate

### CODE REVIEW

File: docs/_includes/floating-label.html
[the remark this answers is not in the document]
Comment (agliga) [reply docs/_includes/floating-label.html:1]: This might be related to makeup

File: docs/_includes/textbox.html [hunk 1]
```diff
@@ -28,14 +28,14 @@ <h3>Readonly Textbox</h3>
...
-            <span class="textbox">
+            <span class="textbox textbox--readonly">
                 <input aria-label="Textbox demo" class="textbox__control" type="text" value="Readonly textbox" readonly />
             </span>
         </div>
     </div>
 
     {% highlight html %}
-<span class="textbox">
+<span class="textbox textbox--readonly">
```
Comment (agliga) [inline docs/_includes/textbox.html:38]: I think we should not use this for all cases but only for icon cases. 
For default we should keep disabled only on button.

File: src/sass/textbox/textbox.scss [hunk 2]
```diff
@@ -3,25 +3,86 @@
...
-        right: 0;
-        top: 0;
     }
 }
 
-span.textbox {
-    display: inline-block;
+/* `.textbox--focus` should be removed after full browser support for `:has` */
+.textbox--focus,
+.textbox:has(> .textbox__control:focus):not(
```
[the remark this answers is not in the document]
Comment (ArtBlue) [reply src/sass/textbox/textbox.scss:38]: Gotcha. Makes sense.

File: docs/_includes/floating-label.html
[the remark this answers is not in the document]
Comment (LuLaValva) [reply docs/_includes/floating-label.html:1]: Credit to @agliga

File: docs/_includes/textbox.html (hunk 1 above)
Comment (LuLaValva) [reply docs/_includes/textbox.html:38]: Discussed, we decided that what I have now is okay for consistency.

File: src/sass/textbox/stories/icon.stories.js [hunk 3]
```diff
@@ -56,7 +56,7 @@ export const dualActionable = () => `
 `;
 
 export const actionableLarge = () => `
-<span class="textbox textbox--icon-end">
+<span class="textbox">
     <input class="textbox__control textbox__control--large" type="text" placeholder="placeholder text" />
```
[the remark this answers is not in the document]
Comment (LuLaValva) [reply src/sass/textbox/stories/icon.stories.js]: Good call out, nice breaking change for skin

File: src/sass/textbox/textbox.scss [hunk 4]
```diff
@@ -3,25 +3,86 @@
...
+    align-items: center;
+    border-style: solid;
+    border-width: 1px;
+    box-sizing: border-box;
+    @include color-token(
+        textbox-foreground-color,
+        color-foreground-on-secondary
+    );
+
+    display: flex;
```
[the remark this answers is not in the document]
Comment (LuLaValva) [reply src/sass/textbox/textbox.scss]: Nice catch

File: src/sass/textbox/textbox.scss [hunk 5]
```diff
@@ -3,25 +3,86 @@
 
 .textbox {
     @include color-token(textbox-foreground-color, color-background-secondary);
+    @include background-color-token(
+        textbox-background-color,
+        color-background-secondary
+    );
+    @include border-color-token(textbox-border-color, color-stroke-default);
```
[the remark this answers is not in the document]
Comment (LuLaValva) [reply src/sass/textbox/textbox.scss:13]: For this we opted to go with the box-shadow approach that we use in phone input

**Review by LuLaValva — COMMENTED** [review]
Thanks for the review @ArtBlue, lots of good finds. Should all be fixed now.

File: src/sass/phone-input/phone-input.scss [hunk 6]
```diff
@@ -10,7 +10,14 @@
     border-radius: 8px 0 0 8px;
     min-width: 64px;
     padding-inline: var(--spacing-200) var(--spacing-100);
-    border-right: none;
+    border-right-style: revert;
+}
+
+.phone-input button.btn:focus-visible {
+    box-shadow: 0 0 0 0.5pt var(--color-stroke-strong);
```
Comment (ArtBlue) [inline src/sass/phone-input/phone-input.scss:17]: `pt` is actually better suited for print. Can we use `px` instead?

File: docs/_includes/floating-label.html
[the remark this answers is not in the document]
Comment (ArtBlue) [reply docs/_includes/floating-label.html:1]: I'm still seeing some weirdness. Is this from Makeup? The floating label is "floated" by default.

<kbd>image</kbd>

File: docs/_includes/floating-label.html [hunk 7]
```diff
@@ -1,431 +1,350 @@
...
-        <input class="textbox__control" id="last-name" type="text" value="Smith" />
+      <input class="textbox__control" id="last-name" type="text" value="Smith" />
     </span>
-</span>
-    {% endhighlight %}
+  </span>
+  {% endhighlight %}
 
-    <h3>Large floating label</h3>
+  <h3>Large floating label</h3>
```
Comment (ArtBlue) [inline docs/_includes/floating-label.html:54]: We typically add a short paragraph indicating the modifier to be used and the impact of the modifier. Can we add that here?

File: docs/_includes/floating-label.html [hunk 8]
```diff
@@ -1,431 +1,350 @@
...
+  {% endhighlight %}
 
-    {% highlight html %}
-<span class="floating-label">
-    <label class="floating-label__label">Select Option</label>
-    <span class="select">
-        <select name="options">
+  <h3>Floating Label with select</h3>
+
+  <div class="demo">
```
Comment (ArtBlue) [inline docs/_includes/floating-label.html:260]: Here too - a short paragraph explaining the code.

File: docs/_includes/floating-label.html [hunk 9]
```diff
@@ -1,431 +1,350 @@
...
-                    <svg class="icon icon--chevron-down-12" height="8" width="8" aria-hidden="true">
-                        {% include symbol.html name="chevron-down-12" %}
-                    </svg>
-                </span>
-            </span>
-        </div>
+  </span>
+  {% endhighlight %}
+
+  <h3>Floating Label with large select</h3>
```
Comment (ArtBlue) [inline docs/_includes/floating-label.html:296]: Here too - a short paragraph explaining the code.

File: src/sass/phone-input/phone-input.scss [hunk 10]
```diff
@@ -10,7 +10,14 @@
     border-radius: 8px 0 0 8px;
     min-width: 64px;
     padding-inline: var(--spacing-200) var(--spacing-100);
-    border-right: none;
+    border-right-style: revert;
```
Comment (ArtBlue) [inline src/sass/phone-input/phone-input.scss:13]: This looks to be causing a display issue where the line is much darker than it used to be. 

<kbd>image</kbd>

File: src/sass/textbox/textbox.scss [hunk 11]
```diff
@@ -39,7 +37,7 @@
         .textbox--readonly,
         :has(> .textbox__control[readonly])
     ) {
-    outline: auto;
+    box-shadow: 0 0 0 0.5pt var(--color-stroke-strong);
```
Comment (ArtBlue) [inline src/sass/textbox/textbox.scss:40]: Let's use `px` here as well.

File: src/sass/date-textbox/stories/date-textbox.stories.js
[the remark this answers is not in the document]
Comment (ArtBlue) [reply src/sass/date-textbox/stories/date-textbox.stories.js:1]: @LuLaValva , story missing here.

**Review by ArtBlue — CHANGES_REQUESTED** [review]
Nice fixes. I found a couple of more things...

File: src/sass/phone-input/phone-input.scss (hunk 10 above)
Comment (LuLaValva) [reply src/sass/phone-input/phone-input.scss]: My fault for only checking in dark mode 🙈. Switched to `none`.

File: src/sass/phone-input/phone-input.scss (hunk 6 above)
Comment (LuLaValva) [reply src/sass/phone-input/phone-input.scss]: I didn't choose `pt`, it's what we had before. I can change to `px` though, seems more sensible.

File: src/sass/textbox/textbox.scss (hunk 11 above)
Comment (LuLaValva) [reply src/sass/textbox/textbox.scss]: Changed to `1px`, see above

File: docs/_includes/floating-label.html
[the remark this answers is not in the document]
Comment (LuLaValva) [reply docs/_includes/floating-label.html:1]: We fixed this in makeup, see [PR #173](https://github.com/makeup/makeup-js/pull/173)

**Review by LuLaValva — COMMENTED** [review]
Thanks again @ArtBlue! Those documentation comments were on unrelated code that I accidentally formatted so I undid that. We should definitely be more descriptive but those changes are out of scope for this PR.

**Review by agliga — APPROVED** [review]

**Comment by ArtBlue** [comment]
@LuLaValva , it's looking good. Two remaining things:

1. Please resolve the conflicts so we can get an accurate final diff.
2. It's still missing the story we should add in `date-textbox.stories.js`. I made a comment about adding that two iterations ago and reiterated it after. For some reason, I can't see the original comment I made.

**Comment by LuLaValva** [comment]
Thanks for pointing out the range again @ArtBlue, apologies for missing it earlier.

**Review by ArtBlue — APPROVED** [review]
Nice, Luke! Looks great!

**Pull request merged**